akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -358,6 +358,20 @@ trait PathMatchers {
 
   /**
    * A PathMatcher that matches and extracts the complete remaining,
+   * unmatched part of the request's URI path as an decoded (with actual whitespace) String.
+   * @group pathmatcher
+   */
+  object RemainingDecoded extends PathMatcher1[String] {
+    def apply(path: Path) = {
+      Segments(RemainingPath(path).extractions._1) match {
+        case Matched(_, extractions) ⇒ Matched(Path.Empty, Tuple1(extractions._1.mkString))
+        case _ ⇒ Unmatched
+      }
+    }
+  }
```

Comment:
The only suitable method for doing what I need was actually found in `JavaUri.scala`:

```
def gatherSegments(path: sm.Uri.Path): List[String] = path match {
      case Empty               ⇒ Nil
      case Segment(head, tail) ⇒ head :: gatherSegments(tail)
      case Slash(tail)         ⇒ gatherSegments(tail)
    }
```

As far as I understand this is actual decoding process as segments store decoded data. But I really do not want to copy paste this method or import `akka.http.impl.model.JavaUri` in `akka.http.scaladsl.server.PathMatcher`.